Src/FluentAssertions/Equivalency/EquivalencyValidator.cs

```diff
@@ -31,38 +31,38 @@ public void RecursivelyAssertEquality(Comparands comparands, IEquivalencyValidat
     {
         var scope = AssertionScope.Current;
 
-        if (ShouldCompareNodesThisDeep(context.CurrentNode, context.Options, scope))
+        if (ShouldContinueThisDeep(context.CurrentNode, context.Options, scope))
         {
-            UpdateScopeWithReportableContext(scope, comparands, context.CurrentNode);
+            TrackWhatWeNeedToProvideContextToFailures(scope, comparands, context.CurrentNode);
 
             if (!context.IsCyclicReference(comparands.Expectation))
             {
-                RunStepsUntilEquivalencyIsProven(comparands, context);
+                TryToProveNodesAreEquivalent(comparands, context);
             }
         }
     }
 
-    private static bool ShouldCompareNodesThisDeep(INode currentNode, IEquivalencyAssertionOptions options,
+    private static bool ShouldContinueThisDeep(INode currentNode, IEquivalencyAssertionOptions options,
         AssertionScope assertionScope)
     {
         bool shouldRecurse = options.AllowInfiniteRecursion || currentNode.Depth <= MaxDepth;
-
         if (!shouldRecurse)
         {
+            // This will throw, unless we're in using (new AssertionScope())
             assertionScope.FailWith($"The maximum recursion depth of {MaxDepth} was reached.  ");
         }
 
         return shouldRecurse;
     }
 
-    private static void UpdateScopeWithReportableContext(AssertionScope scope, Comparands comparands, INode currentNode)
+    private static void TrackWhatWeNeedToProvideContextToFailures(AssertionScope scope, Comparands comparands, INode currentNode)
```

Comment:
I'm unfamiliar with using the more personal `We` in method names, so find it a bit odd.

Two alternatives
`TrackWhatIsNeededToProvideContextToFailures`
`TrackContextForFailureMessage`